lib/stat_tracker.rb

```diff
@@ -23,6 +23,48 @@ def initialize(files)
     end
   end
 
+  #array off all gameteam objects played in that season
+  #can't get from just @games because doesn't have tackles so need to relate
+  #season_id to game id then game_id in gameteams data to game_id there
+  #this can definitely be refactored
+  def games_played_in_season(season)
+    game_objects_by_season = @games.group_by do |game|
+      game.season
+    end
+    get_games_in_season = game_objects_by_season.keep_if do |key, value|
+      key == season
+    end
+    season_game_ids = get_games_in_season.flat_map do |season, games|
+      games.flat_map {|game| game.game_id}
+    end
+    @game_teams.find_all {|games| season_game_ids.include?(games.game_id)}
+  end
+  #Hash of total team tackles by team_id based ion the previous helper method of\
+  #games_played_in_season
+  def total_tackles_by_team_id(season)
+    games_played_in_season(season).each_with_object(Hash.new(0)) do |gameteam, hash|
```

Comment:
I like the helper methods! These keep the main methods #most_tackles && #fewest_tackles clean to read.